cmake/Dependencies.common.cmake

```diff
@@ -187,3 +187,16 @@ endif()
 set(CUTLASS_ENABLE_HEADERS_ONLY ON CACHE BOOL "Enable only the header library")
 check_and_add_cmake_submodule(${PROJECT_SOURCE_DIR}/third_party/cutlass)
 list(APPEND DALI_LIBS CUTLASS)
+
+##################################################################
+# CocoAPI
+##################################################################
+set(GENERATE_POSITION_INDEPENDENT_CODE ON CACHE BOOL "-fPIC")
+set(ENABLE_SHARED OFF CACHE BOOL "shared library target")
+set(ENABLE_STATIC ON CACHE BOOL "static library target")
+
+project(cocoapi)
+set(SOURCE_FILES third_party/cocoapi/common/maskApi.c)
+add_library(cocoapi STATIC ${SOURCE_FILES})
```

Comment:
I would add:
```
target_link_libraries(cocoapi PRIVATE "-pie")
set_target_properties(cocoapi PROPERTIES POSITION_INDEPENDENT_CODE ON)
``

Reply:
OK I can try that. It obviously builds on my machine so don't know how to test it though.

> I would add:
> 
> ```
> target_link_libraries(cocoapi PRIVATE "-pie")
> set_target_properties(cocoapi PROPERTIES POSITION_INDEPENDENT_CODE ON)
> ``
> ```

It does not build anymore on my machine if I add the 
```
-pie
```
option, it makes GCC thinks I'm building an executable not a library.

Did you mean -fpic ?

Reply:
You are right, I have confused it with the part where we are building the tests.
I see that we set `PIC` project wide in https://github.com/NVIDIA/DALI/blob/master/CMakeLists.txt#L140. So maybe if you remove this `project(cocoapi)` it will just work?

Reply:
I'm able to reproduce on my machine, I will take a look and come back later.

Reply:
It looks like it's enough to add 
```
set_target_properties(cocoapi PROPERTIES POSITION_INDEPENDENT_CODE ON)``` to make it compile. I also needed to adjust the test to the changes for CocoLoader constructor:

Changing the `dali/operators/reader/loader/loader_test.cc:118` to:
```
    std::vector<int> heights;
    std::vector<int> widths;
    std::vector<int> offsets;
    std::vector<float> boxes;
    std::vector<int> labels;
    std::vector<int> counts;
    std::vector<std::vector<int> > masks_meta;
    std::vector<std::vector<float> > masks_coords;
    std::vector<std::vector<std::string> > masks_rles;
    std::vector<std::vector<int> > masks_rles_idx;
    std::vector<int> original_ids;
    std::string file_root = testing::dali_extra_path() + "/db/coco/images";
    std::string annotations_file = testing::dali_extra_path() + "/db/coco/instances.json";
    auto coco_spec = OpSpec("COCOReader")
                      .AddArg("file_root", file_root)
                      .AddArg("annotations_file", annotations_file)
                      .AddArg("batch_size", 32)
                      .AddArg("device_id", 0)
                      .AddArg("dont_use_mmap", dont_use_mmap);
    shared_ptr<dali::CocoLoader> reader(
        new CocoLoader(
            coco_spec, heights, widths, offsets, boxes, labels, counts, masks_meta,
            masks_coords, masks_rles, masks_rles_idx, false, false, false, original_ids));
```
I was able to build and run the test successfully. Can you add that or similar change there?

Reply:
Sure !